controllers/awsmachine_controller.go

```diff
@@ -798,22 +798,31 @@ func (r *AWSMachineReconciler) reconcileLBAttachment(machineScope *scope.Machine
 		return r.deregisterInstanceFromV2LB(machineScope, elbsvc, i)
 	}
 
-	// This changes the flow because previously it didn't care about this part.
-	if elbScope.ControlPlaneLoadBalancer().LoadBalancerType == infrav1.LoadBalancerTypeClassic {
+	switch elbScope.ControlPlaneLoadBalancer().LoadBalancerType {
+	case infrav1.LoadBalancerTypeClassic:
+		fallthrough
+	case "":
```

Comment:
With the default now set through kube is this even possible?

Reply:
I've documented the corner case in the test, and tried to describe it in the PR summary. What Luther helped me find out is this: I had v2.0.2 CRDs installed, but v2.0.2+X controller running. This led to an `AWSCluster` being stored in etcd without having a value for `loadBalancerType` (i.e. empty string). Since a mutating webhook only runs on create/update, but the controller has no reason to perform such a change, the controller still runs with the object with the empty field value. The solution is therefore easy: support that here. Very likely, only few people will even hit this, but the chances exist. Having the wrong CRD versions installed doesn't even have to be a human error – it could even happen when some "crd-install" job technically fails but the new controller version starts up anyway. It's just a risk that we should cover.

Reply:
Right. I just thought that the annotation might have fixed that. But I see now. :)